File: DSAlgorithms/patterns/2_pointers/2_remove_element.cpp
#include <iostream>
#include <bits/stdc++.h>
class remove_element 
{
    public: 
        static int remove(std::vector<int>& arr, int key)
        {
            int next_element = 0;
            for(int i = 0; i < arr.size(); i++)
            {
                if(arr[i] != key)
                {
                    arr[next_element] = arr[i];
                    next_element++;
                }
            }
            return next_element;
        }
};
int main(int argc, char* argv[])
{
    std::vector<int> arr = {3, 2, 3, 6, 3, 10, 9, 3};
    std::cout<<"Array new length: "<<remove_element::remove(arr, 3)<<'\n';
    arr = {2, 11, 2, 2, 1};
    std::cout<<"Array new length: "<<remove_element::remove(arr, 2)<<'\n';
}

File: dsa_algo/patterns/2_pointers/2_remove_element.ts
const remove_element = (arr: number[], key: number) => 
{
    let next_element = 0;
    for(let i = 0; i < arr.length; i++)
    {
        if(arr[i] !== key)
        {
            arr[next_element] = arr[i];
            next_element++
        }
    }
    return next_element;
}
console.log(`Array new length: ${remove_element([3, 2, 3, 6, 3, 10, 9, 3], 3)}`);
console.log(`Array new length: ${remove_element([2, 11, 2, 2, 1], 2)}`);
